internal/manifest/v1alpha1/custom_resource.go

```diff
@@ -0,0 +1,75 @@
+package v1alpha1
+
+import (
+	"context"
+
+	manifestv1alpha1 "github.com/kyma-project/module-manager/api/v1alpha1"
+	declarative "github.com/kyma-project/module-manager/pkg/declarative/v2"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const CustomResourceManager = "resource.kyma-project.io/finalizer"
+
+func PostRunCreateCR(
+	ctx context.Context, skr declarative.Client, kcp client.Client, obj declarative.Object,
+) error {
+	manifest := obj.(*manifestv1alpha1.Manifest)
+	resource := manifest.Spec.Resource.DeepCopy()
+	if resource.Object == nil {
+		return nil
+	}
+
+	if err := skr.Create(
+		ctx, resource, client.FieldOwner(CustomResourceManager),
+	); err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	oMeta := &v1.PartialObjectMetadata{}
+	oMeta.SetName(obj.GetName())
+	oMeta.SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
+	oMeta.SetNamespace(obj.GetNamespace())
+	oMeta.SetFinalizers(obj.GetFinalizers())
+	if added := controllerutil.AddFinalizer(oMeta, CustomResourceManager); added {
+		if err := kcp.Patch(
+			ctx, oMeta, client.Apply, client.ForceOwnership, client.FieldOwner(CustomResourceManager),
+		); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func PreDeleteDeleteCR(
```

Comment:
could we  add a comment to this function to quickly understand which resource is getting deleted?

Reply:
I will add comments to both functions reflecting their use case for the Manifest